proto/storage/notification.proto

```diff
@@ -12,8 +12,8 @@ package storage;
 //
 // Refer to v1.Notification for a more detailed doc.
 message Notification {
-    // This is a UUIDv5 where the tuple of domain, resource_type, resource_id, message is used to generate it
-    // deterministically. This can be then used for deduping purposes.
+    // The id is a UUIDv5 generated deterministically from the tuple (type, level, domain,
+    // resource_type, resource_id, message). It is used for deduplication of notifications.
     string id                               = 1 [(gogoproto.moretags) = 'sql:"pk,type(uuid)"'];
     NotificationType type                   = 2 [(gogoproto.moretags) = 'search:"Notification Type,hidden"'];
     NotificationLevel level                 = 3 [(gogoproto.moretags) = 'search:"Notification Level,hidden"'];
```

Comment:
I've recently learned that if you want to use query framework to build SQL queries on these columns, you also need to:
* update search options in pkg/search/options.go
* add your proto as category in proto/api/v1/search_service.proto to have `GetByQuery()` generated

Reply:
* the options already exist, see https://github.com/stackrox/stackrox/blob/39dd32aaa92999769981505667d56fbd19cf8e97/pkg/search/options.go#L340-L342
* see https://github.com/stackrox/stackrox/pull/7593/files#diff-b66923245dffac530f51a23be56cc482ae00b2b5c9ab01dce715ce97ac3639c2L71-L72.